tensorflow/contrib/receptive_field/python/util/receptive_field_test.py

```diff
@@ -220,6 +246,36 @@ def testComputeRFFromGraphDefNonSquareRF(self):
     self.assertEqual(effective_padding_x, 0)
     self.assertEqual(effective_padding_y, 0)
 
+  def testComputeRFFromGraphDefStopPropagation(self):
+    graph_def = create_test_network_6().as_graph_def()
+    input_node = 'input_image'
+    output_node = 'output'
+    # Compute the receptive field but stop the propagation for the random
+    # uniform variable of the dropout
+    (receptive_field_x, receptive_field_y, effective_stride_x,
+     effective_stride_y, effective_padding_x, effective_padding_y) = (
+         receptive_field.compute_receptive_field_from_graph_def(
+             graph_def, input_node, output_node,
+             ['Dropout/dropout/random_uniform']))
```

Comment:
In this case, I believe the same behavior could be achieved by adding a new op "RandomUniform" to "_UNCHANGED_RF_LAYER_OPS". Correct?
This is just to check my understanding, I am not suggesting you to do this. This is a great test for the stop_propagation option.

Reply:
Almost. The implementation of `dropout` uses the shape of the input features to generate uniform random numbers of matching shape. That introduces a dependency of a `RandomUniform` operation on the input features but there is no meaningful way to propagate the receptive field. So the propagation of the receptive field needs to be stopped rather than passed through unchanged.

Reply:
hmmm... if we added the "RandomUniform" op to "_UNCHANGED_RF_LAYER_OPS", it would still work I guess? Maybe I am missing something here.
If I understand correctly, based on what you said and based on my understanding of dropout, the graph with Dropout will have a branch to generate random values which will be used to elementwise multiply the features, deactivating some of them. So if we simply propagated back the RF through this branch without changing it, the computation would still work?

Reply:
Yes, that's right. I confused myself.

Reply:
thanks, glad I understand this now.